### PR TITLE
docs: add custom objects beta API documentation

### PR DESCRIPTION
**Pull request by shivansh**
https://app.devrev.ai/devrev/works/ISS-121317

The PR will be ready for review as soon as the custom object APIs are beta (in-progress) ✅

### CODE REVIEW

File: fern/docs/pages/custom-objects.mdx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+```
+
+## Key Concepts
+
+1. **Leaf Type**: The base type of your custom object (e.g., "campaign").
+2. **Subtype**: A more specific categorization within a leaf type (e.g., "promotion" or "advertising" for a "campaign" leaf type).
+3. **Custom Fields**: User-defined fields that store specific data for your custom object.
+4. **Unique Key**: A unique identifier for each custom object, useful for maintaining idempotency.
+
+For more details, please refer to the [Customization](./customization.mdx) documentation.
```
Comment (shivansh) [inline fern/docs/pages/custom-objects.mdx:170]: @Atul-Butola please let me know if this isn't the right way to link other docs.

Comment (shivansh) [reply fern/docs/pages/custom-objects.mdx]: I tested this on the published docs preview and it doesn't work.

Comment (shivansh) [reply fern/docs/pages/custom-objects.mdx]: Fixed this.

File: fern/docs/pages/custom-objects.mdx [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+            "field_type": "enum",
+            "description": "Target audience for the campaign",
+            "allowed_values": [
+                "Professionals",
+                "Students"
+            ]
+        }
+    ],
+    "is_custom_leaf_type": true,
+    "id_prefix": "CAMP"
```
Comment (praneethratna) [inline fern/docs/pages/custom-objects.mdx:72]: Can we add definitions regarding `is_custom_leaf_type` and `id_prefix`?

**Review by praneethratna — APPROVED** [review]

File: fern/docs/pages/custom-objects.mdx [hunk 3]
```diff
@@ -0,0 +1,272 @@
+## Table of contents
+
+* [Introduction](#custom-objects)
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:3]: Remove. Fern automatically constructs a TOC from the headings so no need to create one explicitly.

File: fern/docs/pages/custom-objects.mdx [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+   - [Delete a custom object](#delete-a-custom-object)
+* [Key concepts](#key-concepts)
+* [Custom object lifecycle](#custom-object-lifecycle)
+* [Extended use cases](#extended-use-cases)
+    - [Reference custom objects in other objects](#reference-custom-objects-in-other-objects)
+    - [Add subtypes to custom objects](#add-subtypes-to-custom-objects)
+
+- - -
+
+## Introduction
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:18]: Remove.
```suggestion
```

File: fern/docs/pages/custom-objects.mdx [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+Custom objects allow you to extend DevRev's data model beyond the standard use-cases
+served by the native apps like Build and Support.  Custom objects allow you to create
+and manage object types tailored to your specific business needs.
+
+## Quick start
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:23]:
```suggestion
## Custom object example
```

File: fern/docs/pages/custom-objects.mdx [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+curl --location 'https://api.devrev.ai/custom-objects.delete' \
+--header 'Content-Type: application/json' \
+--header 'Accept: application/json' \
+--header 'Authorization: <YOUR_API_TOKEN>' \
+--data '{
+    "id": "don:core:dvrv-us-1:devo/demo:custom_object/campaign/1"
+}'
+```
+
+## Key concepts
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:195]: Move this up to be the first section after the intro lines.

File: fern/docs/pages/custom-objects.mdx [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+
+1. **Leaf Type**: The base type of your custom object (e.g., "campaign"). Note that for custom objects, `is_custom_leaf_type` must be set to `true` to differentiate it from standard DevRev objects.
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:197]:
```suggestion
1. **Leaf type**: The base type of your custom object (for example, "campaign"). For custom objects, `is_custom_leaf_type` must be set to `true` to differentiate it from standard DevRev objects.
```

File: fern/docs/pages/custom-objects.mdx [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+2. **Subtype**: A more specific categorization within a leaf type (e.g., "promotion" or "advertising" for a "campaign" leaf type).
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:198]:
```suggestion
2. **Subtype**: A more specific categorization within a leaf type (for example, "promotion" or "advertising" for a "campaign" leaf type).
```

File: fern/docs/pages/custom-objects.mdx [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+3. **Custom Fields**: User-defined fields that store specific data for your custom object.
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:199]:
```suggestion
3. **Custom fields**: User-defined fields that store specific data for your custom object.
```

File: fern/docs/pages/custom-objects.mdx [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+4. **Unique Key**: The `unique_key` is a unique identifier for each custom object. This is useful for maintaining idempotency when retrying custom object creation requests.
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:200]:
```suggestion
4. **Unique key**: A unique identifier for each custom object. This is useful for maintaining idempotency when retrying custom object creation requests.
```

File: fern/docs/pages/custom-objects.mdx [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+5. **ID Prefix**: The `id_prefix` should be unique and is used to generate the display ID for the custom object. In the above example, the `id_prefix` is "CAMP" and the generated custom object display ID is "C-CAMP-1". The display ID is used to identify the custom object in the UI, similar to the standard DevRev object display IDs like "ISS-001" for issues.
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:201]:
```suggestion
5. **ID prefix**: Used to generate the display ID for the custom object. In the above example, the `id_prefix` is "CAMP" and the generated custom object display ID is "C-CAMP-1". The display ID is used to identify the custom object in the UI, similar to the standard DevRev object display IDs like "ISS-001" for issues.
```

File: fern/docs/pages/custom-objects.mdx [hunk 12]
```diff
@@ -0,0 +1,272 @@
...
+
+For more details on customization concepts, please refer to the
+[Customization](./object-customization) documentation.
+
+## Custom object lifecycle
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:206]: Move the contents of this section up to the lifecycle operations reference.

File: fern/docs/pages/custom-objects.mdx [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+Custom objects allow you to extend DevRev's data model beyond the standard use-cases
+served by the native apps like Build and Support.  Custom objects allow you to create
+and manage object types tailored to your specific business needs.
+
+## Quick start
+Let's say you want to manage marketing campaigns on DevRev. We'll go through the process of creating a "Campaign" custom object with relevant fields.
+
+All DevRev objects require a schema. So, we'll first create a schema for the "Campaign" custom object. Make sure to replace the `<YOUR_API_TOKEN>` with your actual API token.
+
+### Create a schema and a custom object
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:28]: Insert an H2 here
```suggestion
## Custom object lifecycle

### Create a schema and a custom object
```

File: fern/docs/pages/custom-objects.mdx [hunk 14]
```diff
@@ -0,0 +1,272 @@
...
+For more details on customization concepts, please refer to the
+[Customization](./object-customization) documentation.
+
+## Custom object lifecycle
+
+1. **Creation**: Define the leaf type, subtype (optional), and custom fields for your custom object.
+2. **Usage**: Create, update, and query custom objects as needed in your workflows.
+3. **Management**: Modify the schema of the custom object as your needs evolve.
+
+## Extended use cases
```
Comment (bc-devrev) [inline fern/docs/pages/custom-objects.mdx:212]: Is there another description you could use besides "extended"?

**Review by bc-devrev — CHANGES_REQUESTED** [review]

File: fern/docs/pages/custom-objects.mdx (hunk 5 above)
Comment (shivansh) [reply fern/docs/pages/custom-objects.mdx:23]: How about "Getting started"? "Custom object example" doesn't fit well with the flow of the doc.

File: fern/docs/pages/custom-objects.mdx (hunk 14 above)
Comment (shivansh) [reply fern/docs/pages/custom-objects.mdx:212]: How about "Beyond basic use cases".
Some other alternatives:
1. Advanced Scenarios
1. Advanced Applications
1. Complex Use Cases
1. Advanced Implementations
1. Specialized Configurations
1. Advanced Features
1. Power User Techniques
1. Advanced Customization
1. Beyond Basic Usage
1. Advanced Functionality

Comment (bc-devrev) [reply fern/docs/pages/custom-objects.mdx:190]: Extended, advanced, and beyond basic all have the same meaning, which is nothing. So if there isn't a term that's meaningful I guess you can leave it as-is.

**Review by bc-devrev — APPROVED** [review]

**Pull request merged**